### PR TITLE
[improve][broker] Introduce resourcegroup rate limit on topic and geo

### PR DESCRIPTION
**Pull request by nodece**
### Modifications

- Add dispatch rate limit on topic and geo
- Add resource group bindings to the topic

### Documentation

- [ ] `doc` 
- [x] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java [hunk 1]
```diff
@@ -3928,5 +3928,61 @@ public void removeSchemaCompatibilityStrategy(
...
+
+    @GET
+    @Path("/{tenant}/{namespace}/{topic}/resourceGroup")
+    @ApiOperation(value = "Get ResourceGroup for a topic")
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Topic doesn't exist"),
+            @ApiResponse(code = 405, message =
+                    "Topic level policy is disabled, enable the topic level policy and retry"),
+            @ApiResponse(code = 409, message = "Concurrent modification")
```
[the remark this answers is not in the document]
Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java:3967]: At the same time, resources are being modified by multiple users.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java [hunk 2]
```diff
@@ -80,6 +81,8 @@ public AbstractReplicator(Topic localTopic, String replicatorPrefix, String loca
         this.producer = null;
         this.producerQueueSize = brokerService.pulsar().getConfiguration().getReplicationProducerQueueSize();
 
+        this.replicatorId = String.format("%s | %s", topicName, localCluster + "-->" + remoteCluster);
```
[the remark this answers is not in the document]
Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java:84]: This `replicatorId` sames with the master branch.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 3]
```diff
@@ -1120,31 +1127,56 @@ public void updateResourceGroupLimiter(Optional<Policies> optPolicies) {
...
             if (resourceGroup != null) {
+                TopicName topicName = TopicName.get(topic);
+                resourceGroupService.unRegisterTopic(topicName);
+                String topicRg = topicPolicies.getResourceGroupName().getTopicValue();
+                if (topicRg != null) {
+                    try {
+                        resourceGroupService.registerTopic(topicRg, topicName);
+                    } catch (Exception e) {
+                        log.error("Failed to register resource group {} for topic {}", rgName, topic);
+                        return;
```
[the remark this answers is not in the document]
Comment (nodece) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:1154]: When the resource group does not exist or this topic is already registered to this resource group, an exception will be thrown, so we don't need to do anything.

**Comment by nodece** [comment]
Ping @liudezhi2098

**Review by liudezhi2098 — APPROVED** [review]

**Pull request merged**